exec/exec.go

```diff
@@ -98,17 +101,25 @@ func New() Interface {
 
 // Command is part of the Interface interface.
 func (executor *executor) Command(cmd string, args ...string) Cmd {
```

Comment:
is what we're going for masking ErrDot? would something like this work?

```go
result := osexec.Command(cmd, args...)
result.Err = maskErrDot(result.Err)
return (*cmdWrapper)(result)
```

Where the 1.18 impl just returns the err as-is, and 1.19+ returns nil if `errors.Is(err, exec.ErrDot)`?

Then we could use the same `err = maskErrDot(err)` call in our LookPath impl

Reply:
that would also let us drop the string error comparison

Reply:
on it!

Reply:
Hmm, i can't refer to `result.Err` in the exec.go as `Err` is not present in go1.18. Any reference to `Err` has to be in the file that has the build constraints.

Reply:
ah, ok. I guess two version-defined methods would work... `maskErrDot(err)` and `maskErrDotCmd(cmd)`, etc

Reply:
yep, let me do that

Reply:
Done!